### PR TITLE
clippy fixes

### PR DESCRIPTION
**Pull request by matthiaskrgr**
clippy::redundant_closure
clippy::unnecessary_mut_passed
clippy::single_char_pattern
clippy::clone_on_copy
clippy::into_iter_on_ref
clippy::extra_unused_lifetimes

### CODE REVIEW

File: src/stacked_borrows.rs [hunk 1]
```diff
@@ -684,9 +684,9 @@ impl Stacks {
         state: &GlobalState,
     ) -> InterpResult<'tcx> {
         trace!("deallocation with tag {:?}: {:?}, size {}", tag, alloc_id, range.size.bytes());
-        let mut state = state.borrow_mut();
+        let state = state.borrow_mut();
```
Comment (RalfJung) [inline src/stacked_borrows.rs:687]: Does this mean we can just `borrow` here?

**Comment by RalfJung** [comment]
Thanks! These all look good to me.

What is the clippy invocation you have been using? Might make sense to add this to CI (similar to the rustfmt job, i.e., ignored by bors but displayed in the PR).

File: src/stacked_borrows.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/stacked_borrows.rs:687]: ah that works, too, I'll fix it :smile:

**Comment by matthiaskrgr** [comment]
Just `cargo clippy --all-target --all-features`.
However I left these warnings untouched because IMO they are a bit "meh":
```
warning: you are deriving `PartialEq` and can implement `Eq`
  --> src/eval.rs:22:30
   |
22 | #[derive(Copy, Clone, Debug, PartialEq)]
   |                              ^^^^^^^^^ help: consider deriving `Eq` as well: `PartialEq, Eq`
   |
   = note: `#[warn(clippy::derive_partial_eq_without_eq)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq

warning: you are deriving `PartialEq` and can implement `Eq`
  --> src/eval.rs:32:30
   |
32 | #[derive(Copy, Clone, Debug, PartialEq)]
   |                              ^^^^^^^^^ help: consider deriving `Eq` as well: `PartialEq, Eq`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq

warning: `miri` (lib test) generated 2 warnings (2 duplicates)
warning: `miri` (lib) generated 2 warnings
warning: manual implementation of an assign operation
  --> benches/helpers/fibonacci_helper_iterative.rs:12:9
   |
12 |         b = c + b;
   |         ^^^^^^^^^ help: replace it with: `b += c`
   |
   = note: `#[warn(clippy::assign_op_pattern)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
```

**Comment by RalfJung** [comment]
> However I left these warnings untouched because IMO they are a bit "meh":

Yeah, we can add `allow`s for that. I'll look into the CI config. Thanks!

@bors r+

**Pull request merged**